diff -u libapache2-mod-vhs-ows-1.0.32/mod_vhs.c libapache2-mod-vhs-ows-1.0.32-ows/mod_vhs.c
--- libapache2-mod-vhs-ows-1.0.32/mod_vhs.c	2007-03-07 22:36:03.000000000 +0100
+++ libapache2-mod-vhs-ows-1.0.32-ows/mod_vhs.c	2009-07-28 17:22:07.000000000 +0200
@@ -859,7 +859,7 @@
  */
 typedef struct {
 	int	engine;		// Status of suPHP_Engine
-	chari	*php_config;
+	char	*php_config;
 	int	cmode;		// Server of directory configuration?
 	char	*target_user;
 	char	*target_group;
@@ -896,6 +896,10 @@
 	// Path to the suPHP config file per user
 	char *transformedPath = NULL;
 
+	char *virtualUsername = NULL;
+
+	virtualUsername = apr_pstrcat(r->pool, "#", username, NULL);
+
 	if (vhr->suphp_config_path) {
 		if ((strstr(vhr->suphp_config_path,"%s")!=NULL) && (username!=NULL))
 			transformedPath = apr_psprintf(r->pool, vhr->suphp_config_path, username);
@@ -922,8 +926,9 @@
 
 	cfg->engine       = (strstr(passwd,"engine=Off") == NULL);
 	cfg->php_config   = apr_pstrdup(r->pool,transformedPath);
-	cfg->target_user  = apr_pstrdup(r->pool,"apache");
-	cfg->target_group = apr_pstrdup(r->pool,username);
+	//cfg->target_user  = apr_pstrdup(r->pool,"apache");
+	cfg->target_user  = apr_pstrdup(r->pool,virtualUsername);
+	cfg->target_group = apr_pstrdup(r->pool,"www-data");
 
 	ap_set_module_config(r->server->module_config, suphp_module, cfg);
 }
@@ -1157,6 +1162,7 @@
 #ifdef WANT_VH_HOST
 	apr_table_set(r->subprocess_env, "VH_HOST", host);
 #endif				/* WANT_VH_HOST */
+	apr_table_set(r->subprocess_env, "VH_HOST", p->pw_name ? p->pw_name : "");
 	apr_table_set(r->subprocess_env, "VH_GECOS", p->pw_gecos ? p->pw_gecos : "");
 	/* Do we have handle vhr_Path_Prefix here ? */
 	if (vhr->path_prefix) {
